From 12eb0c9496e802bad9e5ec71cde888b8fdb1b0b4 Mon Sep 17 00:00:00 2001
From: Jiri Denemark <jdenemar@redhat.com>
Date: Mon, 18 May 2020 20:55:42 +0200
Subject: [PATCH] cpu_map: Add pschange-mc-no bit in IA32_ARCH_CAPABILITIES MSR

Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
Reviewed-by: Pavel Hrdina <phrdina@redhat.com>

Origin: backport, https://libvirt.org/git/?p=libvirt.git;a=commit;h=12eb0c94
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1887490
Last-Update: 2020-10-14
X-Backport-Note:
  - some cpu types didn't exist yet, skipped those hunks
  - qemu 5.0/5.1 didn't exist yet, skipped those hunks

---
 src/cpu_map/x86_features.xml                                   | 3 +++
 tests/cputestdata/x86_64-cpuid-Cooperlake-guest.xml            | 1 +
 tests/cputestdata/x86_64-cpuid-Cooperlake-host.xml             | 1 +
 tests/cputestdata/x86_64-cpuid-Cooperlake-json.xml             | 1 +
 tests/cputestdata/x86_64-cpuid-Core-i7-8550U-enabled.xml       | 2 +-
 tests/cputestdata/x86_64-cpuid-Core-i7-8550U-guest.xml         | 1 +
 tests/cputestdata/x86_64-cpuid-Core-i7-8550U-host.xml          | 1 +
 tests/cputestdata/x86_64-cpuid-Core-i7-8550U-json.xml          | 1 +
 .../cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-enabled.xml | 2 +-
 tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-guest.xml | 1 +
 tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-host.xml  | 1 +
 tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-json.xml  | 1 +
 tests/cputestdata/x86_64-cpuid-Xeon-Platinum-9242-enabled.xml  | 2 +-
 tests/cputestdata/x86_64-cpuid-Xeon-Platinum-9242-json.xml     | 1 +
 tests/domaincapsdata/qemu_4.2.0-q35.x86_64.xml                 | 1 +
 tests/domaincapsdata/qemu_4.2.0.x86_64.xml                     | 1 +
 tests/domaincapsdata/qemu_5.0.0-q35.x86_64.xml                 | 1 +
 tests/domaincapsdata/qemu_5.0.0.x86_64.xml                     | 1 +
 tests/domaincapsdata/qemu_5.1.0-q35.x86_64.xml                 | 1 +
 tests/domaincapsdata/qemu_5.1.0.x86_64.xml                     | 1 +
 20 files changed, 22 insertions(+), 3 deletions(-)

--- a/src/cpu_map/x86_features.xml
+++ b/src/cpu_map/x86_features.xml
@@ -509,6 +509,9 @@
   <feature name='mds-no'>
     <msr index='0x10a' edx='0x00000000' eax='0x00000020'/>
   </feature>
+  <feature name='pschange-mc-no'>
+    <msr index='0x10a' edx='0x00000000' eax='0x00000040'/>
+  </feature>
   <feature name='tsx-ctrl'>
     <msr index='0x10a' edx='0x00000000' eax='0x00000080'/>
   </feature>
--- a/tests/cputestdata/x86_64-cpuid-Core-i7-8550U-enabled.xml
+++ b/tests/cputestdata/x86_64-cpuid-Core-i7-8550U-enabled.xml
@@ -5,5 +5,5 @@
   <cpuid eax_in='0x00000007' ecx_in='0x00' eax='0x00000000' ebx='0x009c47ab' ecx='0x00000004' edx='0xac000400'/>
   <cpuid eax_in='0x0000000d' ecx_in='0x01' eax='0x0000000f' ebx='0x00000000' ecx='0x00000000' edx='0x00000000'/>
   <cpuid eax_in='0x80000001' ecx_in='0x00' eax='0x00000000' ebx='0x00000000' ecx='0x00000121' edx='0x2c100800'/>
-  <msr index='0x10a' edx='0x00000000' eax='0x00000008'/>
+  <msr index='0x10a' edx='0x00000000' eax='0x00000048'/>
 </cpudata>
--- a/tests/cputestdata/x86_64-cpuid-Core-i7-8550U-guest.xml
+++ b/tests/cputestdata/x86_64-cpuid-Core-i7-8550U-guest.xml
@@ -26,6 +26,7 @@
   <feature policy='require' name='pdpe1gb'/>
   <feature policy='require' name='invtsc'/>
   <feature policy='require' name='skip-l1dfl-vmentry'/>
+  <feature policy='require' name='pschange-mc-no'/>
   <feature policy='disable' name='hle'/>
   <feature policy='disable' name='rtm'/>
 </cpu>
--- a/tests/cputestdata/x86_64-cpuid-Core-i7-8550U-host.xml
+++ b/tests/cputestdata/x86_64-cpuid-Core-i7-8550U-host.xml
@@ -27,4 +27,5 @@
   <feature name='pdpe1gb'/>
   <feature name='invtsc'/>
   <feature name='skip-l1dfl-vmentry'/>
+  <feature name='pschange-mc-no'/>
 </cpu>
--- a/tests/cputestdata/x86_64-cpuid-Core-i7-8550U-json.xml
+++ b/tests/cputestdata/x86_64-cpuid-Core-i7-8550U-json.xml
@@ -14,6 +14,7 @@
   <feature policy='require' name='xsaves'/>
   <feature policy='require' name='pdpe1gb'/>
   <feature policy='require' name='skip-l1dfl-vmentry'/>
+  <feature policy='require' name='pschange-mc-no'/>
   <feature policy='disable' name='hle'/>
   <feature policy='disable' name='rtm'/>
 </cpu>
--- a/tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-enabled.xml
+++ b/tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-enabled.xml
@@ -6,5 +6,5 @@
   <cpuid eax_in='0x0000000d' ecx_in='0x01' eax='0x0000000f' ebx='0x00000000' ecx='0x00000000' edx='0x00000000'/>
   <cpuid eax_in='0x80000001' ecx_in='0x00' eax='0x00000000' ebx='0x00000000' ecx='0x008003f7' edx='0x2e500800'/>
   <cpuid eax_in='0x80000008' ecx_in='0x00' eax='0x00000000' ebx='0x03001201' ecx='0x00000000' edx='0x00000000'/>
-  <msr index='0x10a' edx='0x00000000' eax='0x00000029'/>
+  <msr index='0x10a' edx='0x00000000' eax='0x00000069'/>
 </cpudata>
--- a/tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-guest.xml
+++ b/tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-guest.xml
@@ -25,4 +25,5 @@
   <feature policy='require' name='rdctl-no'/>
   <feature policy='require' name='skip-l1dfl-vmentry'/>
   <feature policy='require' name='mds-no'/>
+  <feature policy='require' name='pschange-mc-no'/>
 </cpu>
--- a/tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-host.xml
+++ b/tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-host.xml
@@ -26,4 +26,5 @@
   <feature name='rdctl-no'/>
   <feature name='skip-l1dfl-vmentry'/>
   <feature name='mds-no'/>
+  <feature name='pschange-mc-no'/>
 </cpu>
--- a/tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-json.xml
+++ b/tests/cputestdata/x86_64-cpuid-Ryzen-9-3900X-12-Core-json.xml
@@ -20,5 +20,6 @@
   <feature policy='require' name='rdctl-no'/>
   <feature policy='require' name='skip-l1dfl-vmentry'/>
   <feature policy='require' name='mds-no'/>
+  <feature policy='require' name='pschange-mc-no'/>
   <feature policy='disable' name='monitor'/>
 </cpu>
--- a/tests/domaincapsdata/qemu_4.2.0-q35.x86_64.xml
+++ b/tests/domaincapsdata/qemu_4.2.0-q35.x86_64.xml
@@ -47,6 +47,7 @@
       <feature policy='require' name='pdpe1gb'/>
       <feature policy='require' name='invtsc'/>
       <feature policy='require' name='skip-l1dfl-vmentry'/>
+      <feature policy='require' name='pschange-mc-no'/>
     </mode>
     <mode name='custom' supported='yes'>
       <model usable='yes'>qemu64</model>
--- a/tests/domaincapsdata/qemu_4.2.0.x86_64.xml
+++ b/tests/domaincapsdata/qemu_4.2.0.x86_64.xml
@@ -46,6 +46,7 @@
       <feature policy='require' name='pdpe1gb'/>
       <feature policy='require' name='invtsc'/>
       <feature policy='require' name='skip-l1dfl-vmentry'/>
+      <feature policy='require' name='pschange-mc-no'/>
     </mode>
     <mode name='custom' supported='yes'>
       <model usable='yes'>qemu64</model>
